dbos-rules.ts

```diff
@@ -673,35 +685,32 @@ function analyzeRootNode(eslintNode: EslintNode, eslintContext: EslintContext) {
 
   try {
     if (Node.isStatemented(tsMorphNode)) {
+      // TODO: just analyze the statements instead
       tsMorphNode.getFunctions().forEach(analyzeFunction);
       tsMorphNode.getClasses().forEach(analyzeClass);
     }
     else {
-      const fs = require("fs"), path = require("path");
-
-      let localPackageJson;
-
-      try {localPackageJson = require("../package.json");}
-      catch {
-        try {localPackageJson = require("./package.json");}
-        catch {panic("Hoped to find a local package.json. A statemented root node error occured, but could not proceed because this file is missing");}
-      }
-
-      const possibleConflicts = ["typescript-eslint", "@typescript-eslint/utils", "@typescript-eslint/parser", "@typescript-eslint/eslint-plugin"];
+      const possibleConflicts = [
+        "typescript",
+        "typescript-eslint", "@typescript-eslint/utils",
+        "@typescript-eslint/parser", "@typescript-eslint/eslint-plugin"
+      ];
 
       let accumError = "";
 
       for (const possibleConflict of possibleConflicts) {
         const packageJsonPath = path.join(__dirname, "../../../", possibleConflict, "package.json");
 
         if (fs.existsSync(packageJsonPath)) {
-          accumError += `> You installed ${possibleConflict}, version ${require(packageJsonPath).version} (but the plugin needs ${localPackageJson.dependencies[possibleConflict]}).\n`;
+          const userInstalled = readJSON(packageJsonPath).version;
+          const needed = packageJsonInfo.peerDependencies[possibleConflict];
```

Comment:
If we config `peerDependencies` correctly, do we still need this check?

Reply:
In some cases, installing only gives you a warning, rather than an error, for `peerDependencies`; so this is just a failsafe in case someone somehow manages to get the versions messed up and runs the plugin.